### ISSUE TITLE
Option "gic_version=3" raises several spurious interrupts

### ISSUE THREAD

**Issue by MatteoBertolino92** [NienfengYao/armv8-bare-metal#2]
Hello, if I run with the string:
`	 qemu-system-aarch64 -machine virt,gic_version=3 -cpu cortex-a57 -nographic -kernel kernel.elf`
Several exceptions are raised.

I have tried to print the content of ICC_IAR1_EL1 register in order to understand WHICH exception, by adding to aarch64.* the following header and function:
uint32_t raw_read_icc_iar1_el1(void)
```
{
	uint32_t icc_iar1_el1 = 0;
	__asm__ __volatile__("mrs %0, s3_0_c12_c12_0\n\t" : "=r" (icc_iar1_el1) :  : "memory");
	return icc_iar1_el1; 
}
```
The output did surprise me (the following lines are repeated several times:
```
An exception occurs:
exc_type: 0x00000000 00000011
SP: 0x00000000 40FFFF90
Exception Handler! (
	Current EL = 0x00000000 00000004	exc_type : 0x00000000 00000011
	ICC IAR1 EL1 = 0x00000000 000003FF
AARCH64_EXC_SYNC_SPX)
```
So the INTID of the exception is 3FF, basically 1023, so a SPURIOUS interrupt. A spurious interrupt means that there are no enabled INTIDs in the pending state, or all INTIDs in that pending are of insufficient priority to be taken. A typical scenario, according to ARM's guide, is that a spurious interrupt is generated when polling the IARs, this value indicates that there are no interrupts to available to acknowledge.

Basically, I have 2 questions:
1) Why does option ",gic_version=3" lead to several spurious interrupt? (I need this option or I can't print ICC_IAR1_EL1 register).
2) Why in the output "exc_type" is different by the content of ICC_IAR1_EL1? 

Thanks

**Comment by MatteoBertolino92** [comment]
This is actually a gicv2 and not a gicv3. For this reason, it raises several spurious interrupts (INTID = 1023). 
For instance, GICC registers can't be accessed with their architectural name. This is only an example to access BPR0_EL1:
uint32_t raw_read_gic_gicc_bpr0_el1(void) {
   	uint64_t val;
    	__asm__ __volatile__("mrs %0, s3_0_c12_c8_3" : "=r" (val));
    	return val;
}

So I strongly suggest to modify the name of the folder: this is a v2 and not a v3!
